File: src/test/java/com/line/review/episode6/ContactProcessorBadTest.java
package com.line.review.episode6;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.assertj.core.api.Assertions.assertThat;

/**
 * ContactProcessorBad 테스트
 */
class ContactProcessorBadTest {
    
    private ContactProcessorBad processor;
    
    @BeforeEach
    void setUp() {
        processor = new ContactProcessorBad();
    }
    
    @Test
    void shouldReturnNormalizedName_WhenContactIsFriendPerson() {
        // Given
        ContactModel.Person friend = new ContactModel.Person("친구😊", true);
        
        // When
        ReturnValue result = processor.processContact(friend);
        
        // Then
        assertThat(result).isNotNull();
        assertThat(result.getValue()).isEqualTo("친구"); // 이모지가 제거됨
    }
    
    @Test
    void shouldReturnNull_WhenContactIsNotFriend() {
        // Given
        ContactModel.Person notFriend = new ContactModel.Person("친구아님", false);
        
        // When
        ReturnValue result = processor.processContact(notFriend);
        
        // Then
        assertThat(result).isNull();
    }
    
    @Test
    void shouldReturnNull_WhenContactIsNull() {
        // When
        ReturnValue result = processor.processContact(null);
        
        // Then
        assertThat(result).isNull();
    }
    
    @Test
    void shouldReturnNormalizedName_WhenContactIsFriendPerson_Legacy() {
        // Given
        ContactModel.Person friend = new ContactModel.Person("친구😊", true);
        
        // When
        ReturnValue result = processor.processContactLegacy(friend);
        
        // Then
        assertThat(result).isNotNull();
        assertThat(result.getValue()).isEqualTo("친구"); // 이모지가 제거됨
    }
}

File: src/test/kotlin/com/line/review/episode6/ContactProcessorBadTest.kt
package com.line.review.episode6

import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.assertj.core.api.Assertions.assertThat

/**
 * ContactProcessorBad 테스트 (Kotlin 버전)
 */
class ContactProcessorBadTest {
    
    private lateinit var processor: ContactProcessor
    
    @BeforeEach
    fun setUp() {
        processor = ContactProcessor()
    }
    
    @Test
    fun shouldReturnNormalizedName_WhenContactIsFriendPerson() {
        // Given
        val friend = ContactModel.Person("친구😊", true)
        
        // When
        val result = processor.processContactBad(friend)
        
        // Then
        assertThat(result).isNotNull
        assertThat(result?.value).isEqualTo("친구") // 이모지가 제거됨
    }
    
    @Test
    fun shouldReturnNull_WhenContactIsNotFriend() {
        // Given
        val notFriend = ContactModel.Person("친구아님", false)
        
        // When
        val result = processor.processContactBad(notFriend)
        
        // Then
        assertThat(result).isNull()
    }
    
    @Test
    fun shouldReturnNull_WhenContactIsNull() {
        // When
        val result = processor.processContactBad(null)
        
        // Then
        assertThat(result).isNull()
    }
}

